src/pages/portfolio/ui/assets-table.tsx

```diff
@@ -0,0 +1,305 @@
+
+        <Density compact>
+          <Table>
+            <Table.Thead>
+              <Table.Tr>
+                <Table.Th>Asset</Table.Th>
+                <Table.Th>Balance</Table.Th>
+                <Table.Th>Price</Table.Th>
+                <Table.Th>Value</Table.Th>
+                <Table.Th>Actions</Table.Th>
+              </Table.Tr>
+            </Table.Thead>
+            <Table.Tbody>
+              {Array.from({ length: 6 }).map((_, i) => (
+                <Table.Tr key={i}>
+                  <Table.Td>
+                    <div className='flex items-center gap-2'>
+                      <div className='w-8 h-8'>
+                        <Skeleton />
+                      </div>
+                      <div className='w-20 h-4'>
+                        <Skeleton />
+                      </div>
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-4'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-4'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-4'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='flex gap-2'>
+                      <div className='w-5 h-5'>
+                        <Skeleton />
+                      </div>
+                      <div className='w-5 h-5'>
+                        <Skeleton />
+                      </div>
+                    </div>
+                  </Table.Td>
+                </Table.Tr>
+              ))}
+            </Table.Tbody>
+          </Table>
+        </Density>
+      </Card>
+    </div>
+  );
+};
+
+const NotConnectedNotice = () => {
+  return (
+    <div className='p-6'>
+      <Card>
+        <div className='flex flex-col items-center justify-center h-[400px] gap-4'>
+          <Text color='text.secondary' small>
+            Connect wallet to see your assets
+          </Text>
+        </div>
+      </Card>
+    </div>
+  );
+};
+
+const calculateAssetDistribution = (balances: BalancesResponse[]) => {
+  const validBalances = balances.filter(balance => {
+    const valueView = getBalanceView.optional(balance);
+    const metadata = getMetadataFromBalancesResponse.optional(balance);
+    return valueView && metadata;
+  });
+
+  // Calculate values first
+  const valuesWithMetadata = validBalances.map(balance => {
+    const valueView = getBalanceView.optional(balance);
+    const metadata = getMetadataFromBalancesResponse.optional(balance);
+    const amount = valueView?.valueView.value?.amount;
+    if (!amount || !metadata) {
+      return { value: 0, balance };
+    }
+    const formattedAmount = Number(
+      formatAmount({
```

Comment:
I think there are two failure modes:
- somehow missing amount and metadata, in which case the branch on L134 is hit
- somehow formatting results in a bad object and things fall through

Can we add a hint (warn in log?) that the first branch is hit? and can we defend against `formatAmount` failing as well?

The concern is a silent error snowballing into user anxiety that their balance is missing. I think it would be good to leave some breadcrumbs that the asset display is in an unusual state to avoid confusion about actual loss of funds.

Reply:
Good point - I added explicit error handling, and a small inline error message about the failure case to prevent user mistakenly thinking they are missing funds.
<img width="448" alt="image" src="https://github.com/user-attachments/assets/2396d8ed-a26a-4a49-91e8-6e6b2dd681ed" />